### ISSUE TITLE
Carla Library for ARM64 

### ISSUE THREAD

**Issue by SinanErbezci** [carla-simulator/carla#5845]
Hello everyone,
I'm trying to use Jetson Nano as a client and my desktop pc as the server. But I couldn't get carla library for jetson. When i try to use "make PythonAPI"  i receive error. Also linux_x86_64 egg file doesnt work for ARM64. Can anybody provide me the egg file for aarch64 or give suggestions create my own egg file ? Thank you.

**Comment by jokolaco** [comment]
As far as i see you have to compile everything from scratch for arm architecture. Everything around here is using x86. The Ubuntu and Windows builds are x86. arm is a completely different Instruction set architecture (ISA).

https://en.wikipedia.org/wiki/Comparison_of_instruction_set_architectures

**Comment by SinanErbezci** [comment]
@jokolaco Yes, I think so too.  I searched for aarch64 egg file, but couldn't find anywhere. In one post, someone managed to create the eggfile with "make PythonAPI" command. But I couldn't make it. I want to connect the jetson to my PC with an ethernet cable but this problem stalled the process.

**Comment by jokolaco** [comment]
I am not sure if arm is supported. But if this is the case, you should use on your Jetson Nano the same Ubuntu version like the server. Also you need all the libs. Try to understand the compiler errors and follow the script. There a many dependencies of external sources(e.g. libosm2odr.a, libxerces-c.a) which also needs to be compiled on arm. I would suggest just to use another client on x86, besides the nano is not very powerful so you can easily use a cheap old standard pc, laptop as a client or just your server ;-)

**Comment by SinanErbezci** [comment]
@jokolaco thank you for the responses. I've seen some implementations with ROS and jetson. I'll experiment a little bit.

**Comment by Hakistr** [comment]
Hello,
Is there any prebuilt Python libraries(.egg files etc.) that you can share for aarch64 platform?

**Comment by Follu-co** [comment]
> @jokolaco thank you for the responses. I've seen some implementations with ROS and jetson. I'll experiment a little bit.

Have you successfully deployed Carla on Jetson? Can you tell me how it is achieved?
